src/util/prometheus/Http.cpp

```diff
@@ -0,0 +1,61 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of clio: https://github.com/XRPLF/clio
+    Copyright (c) 2023, the clio developers.
+
+    Permission to use, copy, modify, and distribute this software for any
+    purpose with or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL,  DIRECT,  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+#include <util/prometheus/Http.h>
+
+namespace util::prometheus {
+
+namespace http = boost::beast::http;
+
+namespace {
+
+bool
+isPrometheusRequest(http::request<http::string_body> const& req)
+{
+    return req.method() == http::verb::get && req.target() == "/metrics";
+}
+
+}  // namespace
+
+std::optional<http::response<http::string_body>>
+handlePrometheusRequest(http::request<http::string_body> const& req, bool const isAdmin)
```

Comment:
no need const?

Reply:
Why? `const` shows here that `isAdmin` can't be changed in the body of the function